tests/functional/raw-node/test/GCP/bucket/head.js

```diff
@@ -0,0 +1,78 @@
+const assert = require('assert');
+const { GCP } = require('../../../../../../lib/data/external/GCP');
+const { gcpRequestRetry } = require('../../../utils/gcpUtils');
+const { getRealAwsConfig } =
+    require('../../../../aws-node-sdk/test/support/awsConfig');
+
+const credentialOne = 'gcpbackend';
+
+describe('GCP: HEAD Bucket', () => {
+    let gcpClient;
+    let config;
+
+    before(() => {
+        config = getRealAwsConfig(credentialOne);
+        gcpClient = new GCP(config);
+    });
+
+    describe('without existing bucket', () => {
+        beforeEach(function beforeFn(done) {
+            this.currentTest.bucketName = `somebucket-${Date.now()}`;
+            return done();
+        });
+
+        it('should return 404', function testFn(done) {
+            gcpClient.headBucket({
+                Bucket: this.test.bucketName,
+            }, err => {
+                assert(err);
+                assert.strictEqual(err.statusCode, 404);
+                return done();
+            });
+        });
+    });
+
+    describe('with existing bucket', () => {
+        beforeEach(function beforeFn(done) {
+            this.currentTest.bucketName = `somebucket-${Date.now()}`;
+            process.stdout
+                .write(`Creating test bucket ${this.currentTest.bucketName}\n`);
+            gcpRequestRetry({
+                method: 'PUT',
+                bucket: this.currentTest.bucketName,
+                authCredentials: config.credentials,
+            }, 0, (err, res) => {
```

Comment:
Should check for error here in case `res` is undefined.

Reply:
yeah. will do, didn't catch that.